### PR TITLE
Use SYMFONY_REQUIRE for symfony/symfony variant

### PR DESCRIPTION
**Pull request by franmomu**
Closes https://github.com/sonata-project/dev-kit/issues/1502

Instead of creating the new option, this uses the current configuration.

POC: https://github.com/sonata-project/SonataAdminBundle/pull/7111

It is based on https://github.com/doctrine/DoctrineBundle/blob/09339f7df6fc620ce805f52ca0c7424543d30515/.github/workflows/continuous-integration.yml

### CODE REVIEW

**Comment by jordisala1991** [comment]
I think we should modify the name somehow to distinguish between a normal build and a specific sf version build

File: templates/project/.github/workflows/test.yaml.twig [hunk 1]
```diff
@@ -79,6 +91,10 @@ jobs:
               if: matrix.php-version == '8.0'
               run: composer config platform.php 7.4.99
 
+            - name: "Globally install symfony/flex"
+              if: matrix.symfony-require != ''
+              run: "composer global require --no-progress --no-scripts --no-plugins symfony/flex"
```
Comment (VincentLanglet) [inline templates/project/.github/workflows/test.yaml.twig:96]: Why this should be in the `'8.0' != branch.highestPhpVersion.toString` check ?

File: templates/project/.github/workflows/test.yaml.twig [hunk 2]
```diff
@@ -49,12 +54,19 @@ jobs:
...
                       dependencies: highest
                       allowed-to-fail: false
                       variant: '{{ variant.toString|raw }}'
 {% endfor %}
+{% for variant in branch.variants|filter(v => v.package == 'symfony/symfony') %}
+                    - php-version: '{{ branch.targetPhpVersion.toString }}'
+                      dependencies: highest
+                      allowed-to-fail: false
+                      symfony-require: {{ variant.version }}.*
+                      variant: 'normal'
```
Comment (VincentLanglet) [inline templates/project/.github/workflows/test.yaml.twig:68]: We could keep `'{{ variant.toString|raw }}'`

**Review by VincentLanglet — APPROVED** [review]

File: templates/project/.github/workflows/test.yaml.twig [hunk 3]
```diff
@@ -36,6 +39,8 @@ jobs:
 {% endfor %}
                 dependencies: [highest]
                 allowed-to-fail: [false]
+                symfony-require:
+                    - ""
```
Comment (jordisala1991) [inline templates/project/.github/workflows/test.yaml.twig:43]: can you keep the same structure as the rest of this parameters? Ex: ['']

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**